meta_review/handler.py

```diff
@@ -0,0 +1,499 @@
+        time the participant had impact on the meta-review system'. This
+        is the last time among three things: the last time they created/edited
+        a comment, the last time they did a meta-review, the last time
+        their review received a meta-review.
+        """
+        self.logger.info('start updating last active time of all participants')
+
+        for key, participant in self.participants.items():
+            old_active_time = participant.last_active_at
+
+            # check last time they created/edited a comment
+            for comment in participant.comment_set.all():
+                if participant.last_active_at is None:
+                    participant.last_active_at = comment.created_at
+                if comment.created_at > participant.last_active_at:
+                    participant.last_active_at = comment.created_at
+                if (comment.last_edited_at and
+                        comment.last_edited_at > participant.last_active_at):
+                    participant.last_active_at = comment.last_edited_at
+
+            # check last time they did a meta-review
+            for reaction in participant.give.all():
+                if participant.last_active_at is None:
+                    participant.last_active_at = reaction.created_at
+                if reaction.created_at > participant.last_active_at:
+                    participant.last_active_at = reaction.created_at
+
+            # check last time they received a meta-review
+            for reaction in participant.receive.all():
+                if participant.last_active_at is None:
+                    participant.last_active_at = reaction.created_at
+                if reaction.created_at > participant.last_active_at:
+                    participant.last_active_at = reaction.created_at
+
+            if participant.last_active_at != old_active_time:
+                self.logger.debug('%s last active time changed from %s to %s'
+                                  % (participant.login, old_active_time,
+                                     participant.last_active_at))
+
+    @staticmethod
+    def _is_first_deploy():
+        """
+        Check whether it is the first deploy.
+
+        If the scoring system has not been run before, we need to calculate
+        draft score first, and then weight factor. Finally we use that
+        to recalculate score.
+
+        :return: a bool indicates whether this is the first deploy.
+        """
+        # This query set is empty means it is the first deploy.
+        return not Participant.objects.all()
+
+    def _recalculate_score(self):
+        """
+        Recalculate score of each participant.
+
+        Used only on first deploy. Based on the weight factor derived,
+        recalculate score for all participants and comments.
+        """
+        self.logger.info('recalculate score of each participant')
+        # clear score of all comments and participants
```

Comment:
if you create methods on Comment and Participant, the code is self-documenting and this comment is redundant.